src/hera/workflows/v5/workflow.py

```diff
@@ -125,7 +125,22 @@ def _set_workflows_service(cls, v):
             return WorkflowsService()
         return v
 
+    @validator('kind', pre=True, always=True)
+    def _set_kind(cls, v):
+        if v is None:
+            return cls.__name__  # type: ignore
+        return v
+
     def build(self) -> _ModelWorkflow:
+        templates = []
```

Comment:
Deferred template resultion all the way to the end as well to ensure that we capture all the updates to added objects.